index.js

```diff
@@ -535,6 +535,44 @@ nanomatch.matcher = function matcher(pattern, options) {
 };
 
 /**
+ * Returns an array of matches captured by `pattern` in `string, or `null` if the pattern did not match.
+ *
+ * ```js
+ * var nm = require('nanomatch');
+ * nm.capture(pattern, string[, options]);
+ *
+ * console.log(nm.capture('test/*.js', 'test/foo.js));
+ * //=> ['foo']
+ * console.log(nm.capture('test/*.js', 'foo/bar.css'));
+ * //=> null
+ * ```
+ * @param {String} `pattern` Glob pattern to use for matching.
+ * @param {String} `string` String to match
+ * @param {Object} `options` See available [options](#options) for changing how matches are performed
+ * @return {Boolean} Returns an array of captures if the string matches the glob pattern, otherwise `null`.
+ * @api public
+ */
+
+nanomatch.capture = function(pattern, string, options) {
+  var re = nanomatch.makeRe(pattern, extend({capture: true}, options));
+  var unixify = utils.unixify(options);
+
+  function match() {
+    return function(str) {
+      var match = re.exec(unixify(str));
+      if (!match) {
+        return null;
+      }
+
+      return match.slice(1).filter(Boolean);
```

Comment:
`.filter(Boolean)` removes the empty captures/strings, right?
Doesn't that make this API hard to use?

For instance: given `a/**/c/**/f.txt` then both paths below result in the same capture `["x"]`:
- `a/x/c/f.txt`
- `a/c/x/f.txt`

It's impossible to rebuild the matching path from the capture.
I you kept empty strings, then the first would capture `["x", ""]` and the second `["", "x"]`. The matching paths are very easy to rebuild.

Reply:
Good point. This made sense for my use case (described above), but I can see it being weird for other cases. I'll remove the filter here and move it to my code.